src/vm/eeconfig.cpp

```diff
@@ -789,10 +792,18 @@ HRESULT EEConfig::sync()
         g_fEnableARM = TRUE;
     }
 
-    int forceGCconcurrent = CLRConfig::GetConfigValue(CLRConfig::UNSUPPORTED_gcConcurrent);
-    if ((forceGCconcurrent > 0) || (forceGCconcurrent == -1 && g_IGCconcurrent))
+    bool gcConcurrentWasForced = false;
+#ifdef FEATURE_CORECLR
```

Comment:
> what is the order of evaluation supposed to be here? complus overwrites startup flag which overwrites config?

With my changes, we will first determine whether the startup flag for concurrent GC should be set with this criteria:

- If the COMPlus value is set explicitly, we will respect that (treating 1 as true and anything else as false).
- If not, we will check whether the user has set "System.GC.Concurrent" explicitly to true or false in the config file (in the case of CLI this is the JSON configuration) and respect that.
- If neither of them is set explicitly, we will use the default specified in the ConfigDWORDInfo, which for CoreCLR will be 1 (true). 

If we go through all that and decide concurrent GC should be on, we will treat that as 'force' enabling. So with no special configuration set for concurrent GC and GCStress on, we will have both enabled. You can turn off concurrent GC by either setting "System.GC.Concurrent" to false in your JSON config or setting the COMPlus value to 0 (or anything other than 1).